tfjs-layers/src/layers/nlp/multihead_attention_test.ts

```diff
@@ -298,52 +203,60 @@ describe('MultiHeadAttention', () => {
     expectTensorsNotClose(trainOut, testOut);
   });
 
-  describe('Casual Mask Value', () => {
+  describe('Causal Mask Value', () => {
+    let testLayer: MultiHeadAttention;
+    let maskedQuery: Tensor;
+    let maskedValue: Tensor;
+    let mask: Tensor;
+
+    beforeEach(() => {
+      testLayer = new MultiHeadAttention({numHeads: 2, keyDim: 2});
+      const query = tensor2d([
+        [1, 2, 3, 0, 0], [3, 3, 1, 1, 2], [1, 0, 0, 0, 0]
+      ]);
+      const maskedQueryLayer = new Embedding(
+        {inputDim: 4, outputDim: 8, maskZero: true});
+      maskedQuery = maskedQueryLayer.apply(query) as Tensor;
+      const value = tensor2d([[5, 4, 0], [3, 0, 0], [2, 1, 1]]);
+      maskedValue = new Embedding(
+        {inputDim: 6, outputDim: 8, maskZero: true}).apply(value) as Tensor;
+
+      mask = tensor([
+        Array<boolean[]>(3).fill([true, true, false]).concat(
+          Array<boolean[]>(2).fill([false, false, false])),
+        Array<boolean[]>(5).fill([true, false, false]),
+        [[true, true, true]].concat(
+          Array<boolean[]>(4).fill([false, false, false]))
+      ]);
+    });
+
     /**
      * Test that the value and causal masks are taken into account.
      */
-    function testValueMask(testcaseName: string, useCausalMask: boolean) {
```

Comment:
Refactored to explicitly declare both tests instead of using a loop to declare them.